src/test/java/org/broadinstitute/hellbender/utils/genotyper/AlleleListUtilsUnitTest.java

```diff
@@ -107,41 +129,44 @@ public void testSelfPermutation(final List<Allele> alleles1) {
+            }
+        }
         final AlleleList<Allele> originalAlleleList = new IndexedAlleleList<>(alleles1);
-        final AlleleList<Allele> targetAlleleList = new IndexedAlleleList<>(subsetAlleles);
-        final AlleleListPermutation<Allele> subset = AlleleListUtils.permutation(originalAlleleList,targetAlleleList);
-        if (originalAlleleList.alleleCount() == targetAlleleList.alleleCount())
-            throw new SkipException("no real subset");
+        final AlleleList<Allele> targetRandomAlleleList = new IndexedAlleleList<>(randomSubsetAlleles);
+        final AlleleListPermutation<Allele> subset = AlleleListUtils.permutation(originalAlleleList,targetRandomAlleleList);
+        if (originalAlleleList.alleleCount() == targetRandomAlleleList.alleleCount()) {
+            return;//just return, no skip so that it doesnt polute build logs
+        }
         Assert.assertTrue(subset.isPartial());
         Assert.assertFalse(subset.isNonPermuted());
         Assert.assertEquals(subset.fromSize(), originalAlleleList.alleleCount());
-        Assert.assertEquals(subset.toSize(), targetAlleleList.alleleCount());
-        AlleleListUnitTester.assertAlleleList(originalAlleleList,subset.fromList());
-        AlleleListUnitTester.assertAlleleList(targetAlleleList,subset.toList());
+        Assert.assertEquals(subset.toSize(), targetRandomAlleleList.alleleCount());
+        AlleleListUnitTester.assertAlleleList(originalAlleleList, subset.fromList());
+        AlleleListUnitTester.assertAlleleList(targetRandomAlleleList, subset.toList());
 
-        for (int i = 0; i < targetAlleleList.alleleCount(); i++)
-            Assert.assertEquals(subset.fromIndex(i), originalAlleleList.alleleIndex(targetAlleleList.alleleAt(i)));
+        for (int i = 0; i < targetRandomAlleleList.alleleCount(); i++)
+            Assert.assertEquals(subset.fromIndex(i), originalAlleleList.alleleIndex(targetRandomAlleleList.alleleAt(i)));
 
         for (int j = 0; j < originalAlleleList.alleleCount(); j++) {
             final Allele allele = originalAlleleList.alleleAt(j);
-            Assert.assertEquals(subset.toIndex(j), targetAlleleList.alleleIndex(allele));
+            Assert.assertEquals(subset.toIndex(j), targetRandomAlleleList.alleleIndex(allele));
         }
 
-        Assert.assertTrue(AlleleListUtils.equals(subset, targetAlleleList));
+        Assert.assertTrue(AlleleListUtils.equals(subset, targetRandomAlleleList));
     }
 
     @Test(dataProvider = "singleAlleleListData", dependsOnMethods = {"testAsList","testEquals"})
     public void testShufflePermutation(final List<Allele> alleles1) {
         final AlleleList<Allele> originalAlleleList = new IndexedAlleleList<>(alleles1);
-        if (originalAlleleList.alleleCount() <= 1)
-            throw new SkipException("non-shuffle allele-list");
+        if (originalAlleleList.alleleCount() <= 1) {
+            return; //just return, no skip so that it doesnt polute build logs
```

Comment:
actually, lets just change this comment to something like

 "return because this input is invalid for this test"